src/modules/Lan.rb

```diff
@@ -214,51 +214,45 @@ def isAnyInterfaceDown
       down
     end
 
+    # Checks local configuration if IPv6 is allowed
+    #
+    # return [Boolean] true when IPv6 is allowed
     def readIPv6
       @ipv6 = true
 
-      methods =
-        #         "module" : $[
-        #                 "filelist" : ["ipv6", "50-ipv6.conf"],
-        #                 "filepath" : "/etc/modprobe.d/",
-        #                 "regexp"   : "^[[:space:]]*(install ipv6 /bin/true)"
-        #         ]
-        {
-          "builtin" => {
-            "filelist" => ["sysctl.conf"],
-            "filepath" => "/etc/",
-            "regexp"   => "^[[:space:]]*(net.ipv6.conf.all.disable_ipv6)[[:space:]]*=[[:space:]]*1"
-          }
+      methods = {
+        "builtin" => {
+          "filelist" => ["sysctl.conf"],
+          "filepath" => "/etc",
+          "regexp"   => /^[[:space:]]*(net.ipv6.conf.all.disable_ipv6)[[:space:]]*=[[:space:]]*1/
         }
+      }
 
-      Builtins.foreach(methods) do |which, method|
-        filelist = Ops.get_list(method, "filelist", [])
-        filepath = Ops.get_string(method, "filepath", "")
-        regexp = Ops.get_string(method, "regexp", "")
-        Builtins.foreach(filelist) do |file|
-          filename = Builtins.sformat("%1/%2", filepath, file)
-          if FileUtils.Exists(filename)
-            Builtins.foreach(
-              Builtins.splitstring(
-                Convert.to_string(SCR.Read(path(".target.string"), filename)),
-                "\n"
-              )
-            ) do |row|
-              if Ops.greater_than(
-                Builtins.size(
-                  Builtins.regexptokenize(String.CutBlanks(row), regexp)
-                ),
-                0
-              )
-                Builtins.y2milestone("IPv6 is disabled by '%1' method.", which)
-                @ipv6 = false
-              end
-            end
+      methods.each do |_, method|
+        filelist = method["filelist"] || []
+        filepath = method["filepath"] || ""
+        regexp = method["regexp"] || ""
+
+        filelist.each do |file|
+          filename = "#{filepath}/#{file}"
+          next if !FileUtils.Exists(filename)
+
+          begin
+            lines = (SCR.Read(path(".target.string"), filename) || []).split("\n")
```

Comment:
btw instead of `SCR.Read(path(".target.string"), filename)` you can use `TargetFile.read(filename)` from https://github.com/yast/yast-yast2/blob/master/library/system/src/lib/yast2/target_file.rb